Autos, Dynamics, and more

sharkey300 opened the pull request:
We haven't merged into unstable in a while! This pull request adds too many things to list here, but it's mainly dynamics, autos, and comp swerve.

Sloth4915 committed:
Fix capitalization of bling directory

Sloth4915 committed:
Create DriverCommunication segment

Sloth4915 committed:
Merge remote-tracking branch 'origin/vision' into bling

Sloth4915 committed:
Merge branch 'unstable' into bling

sharkey300 committed:
Revert "Fix capitalization of bling directory"

This reverts commit 4c08ad8b8c07c6d5df04561fbafd5a4efbb9dc52.

Sloth4915 committed:
Add Vision integration to bling + bling works now

Sloth4915 committed:
Update BlingShow.java to be more memory efficient

Instead of using a text file for the bling, it now uses a file composed of many bytes. Instead of storing the in a short[][], it now stores it in a byte[][]. This should use half the memory and, according to my (limited) testing, use only a third of the storage space.

otter502 committed:
Add Coral 1 path pair

Sloth4915 committed:
Add DriverCommunication region check

sharkey300 committed:
Merge branch 'unstable' into vision

Sloth4915 committed:
Remove Bling

Sloth4915 committed:
Remove Bling for real this time

sharkey300 committed:
Merge branch 'unstable' into vision

Sloth4915 committed:
(ominous) Bling has returned.

otter502 committed:
Implement "get reverse" function for paths

otter502 committed:
Add Variable Autos File and moved branch side enum to it

otter502 committed:
Merge branch 'vision' into autos

Sloth4915 committed:
Remove bling again!!!!

Sloth4915 committed:
(more ominous) Bling has returned.

Sloth4915 committed:
Add Bling to RobotContainer.java

Sloth4915 committed:
bling2: electric boogaloo

Sloth4915 committed:
Merge remote-tracking branch 'origin/vision' into bling

Sloth4915 committed:
Add DriverCommunication for Vision in reef

otter502 committed:
Add variable autos

Reverse nummbering setup
Add variable autos file
Add missing paths
Add mirroring functionaility
Add ability to convert field branch into pathing commands

otter502 committed:
Fix bug where Align to Reef selected closest branch instead of desired

otter502 committed:
Change mirroed autos to new system

otter502 committed:
Seperate path command and path getter for individual paths

otter502 committed:
Delete unused autos

otter502 committed:
Merge branch 'unstable' into autos

otter502 committed:
Add comment to explain double mirroring

sharkey300 committed:
Add button to debug controller to toggle the reset heading action on mode switch

This feature is practically needed at competition but is incredibly annoying when testing. By binding this to debug controller, we can disable this feature without worrying about it causing problems at competition.

otter502 committed:
Merge branch 'swerve' into autos

Sloth4915 committed:
Add scrollingRainbow segment to processor region

MarshallTappen committed:
Changes necessary to start testing the elevator.  Added Elastic Dashboard to control the elevator.  Added elevator to robot container.  Changed constants that are zero to explicit zeros in the code and removed the constants for them until they are actually used.

MarshallTappen committed:
Setting I constants to zero because we generally do not use I terms.

MarshallTappen committed:
Added code to put the current elevator position on the SmartDashboard

otter502 committed:
Fix rotational drift when moving

Had to remeasure angle offsets for absolute encoders.

otter502 committed:
Merge branch 'swerve' into autos

otter502 committed:
Modify path end points

otter502 committed:
Bring coral station waypoint closer to station

sharkey300 committed:
Merge branch 'unstable' into elevator

sharkey300 committed:
Log limelight poses to SmartDashboard

otter502 committed:
Merge branch 'vision' into autos

sharkey300 committed:
Add max distance check for using mega tag 1 so vision doesn't add bad poses when at station

sharkey300 committed:
Add complex auto chooser to enable drive team to create custom auto sequences

otter502 committed:
Add detect method to intake subsystem

otter502 committed:
Major bug fix

the error here is that, becuase .plus() wraps the rotation2d between [-pi/2, pi/2] it would cause the arm to swing all the way around, since the motion profile saw the setpoint go from -359 to 0 instead -360, and thought it had to go in the opposite way

otter502 committed:
Modify mechanismRenderer to be more accurate to the CAD

otter502 committed:
Add mechanism renderer for setpoints, current position, and target position

sharkey300 committed:
Merge branch 'unstable' into autos

sharkey300 committed:
Merge branch 'autos' into vision

sharkey300 committed:
Check if pose estimate is valid when determing which PoseEstimationMethod to use

sharkey300 committed:
Change field layout used to the correct one based off Team Update 12

DanTCA1 committed:
Merge branch 'unstable' into elevator

Sloth4915 committed:
Merge branch 'unstable' into bling

Sloth4915 committed:
Merge unstable into bling

DanTCA1 committed:
Edit constants to speed up elevator

sharkey300 committed:
Make X on debug controller disable vision measurements so we can test the inluence of vision

sharkey300 committed:
Add basic elastic layout with Auto Chooser and Vision Diagnostics

SPUD42AG committed:
Get arm working on elevator

otter502 committed:
Merge branch 'arm' into dynamics

otter502 committed:
Merge branch 'elevator' into dynamics

sharkey300 committed:
Comment out port forwarding on limelights so elastic layout downloading works

SPUD42AG committed:
Elevator working

otter502 committed:
Modify Renderer to have correct starting angle

otter502 committed:
Set starting angle of arm to 90

otter502 committed:
Update max elevator height to be accurate compared to cad

otter502 committed:
Update mechanism renderer to reflect cad

otter502 committed:
Code DynamicsSystem to Naomi's specifications

This currently works in sim and has to have some changes made to work irl

otter502 committed:
Remove Todo's

otter502 committed:
Add exception in case dynamics (in it's current state) is run irl

otter502 committed:
Merge branch 'dynamics' into autos

otter502 committed:
Merge branch 'autos' of https://github.com/Spartronics4915/2025-Reefscape into autos

otter502 committed:
Add BranchHeight to complex auto chooser

otter502 committed:
Add DynaPreset to enum, remove L2 and L1 options

sharkey300 committed:
Update elastic dashboard layout

otter502 committed:
Flip renderer for ascope to render properly on 3d field

otter502 committed:
Merge branch 'autos' of https://github.com/Spartronics4915/2025-Reefscape into autos

otter502 committed:
Add funnel laserCan

otter502 committed:
Update VariableAutos to include the new dynamics system

otter502 committed:
make Funnel laser can ID match control map

otter502 committed:
Add some safety code / todos to dynamics system

otter502 committed:
Add check if the currently selected auto is the variable autos

otter502 committed:
Change choosen command detection to update on change

otter502 committed:
Remove unneeded assignment from robotContainer

otter502 committed:
Add static instancer to mechanism renderer, switch to using static method in robotcontainer

sharkey300 committed:
Add boolean to vision indicating if inital pose has been set

This is also published to SmartDashboard

sharkey300 committed:
Merge branch 'vision' into autos

sharkey300 committed:
Update elastic layout to show if complex auto is selected and if inital pose is set

otter502 committed:
Rename constant to match with conventions

otter502 committed:
Sync the readme, actual controls, and button map

otter502 committed:
Add funnel auto intake

Sloth4915 committed:
Fix orientTowardsReef to point in right direction

Sloth4915 committed:
Merge branch 'autos' into bling

Sloth4915 committed:
Add getClosestRegion method in DriverCommunication

otter502 committed:
add comp_chassis constants folder

sharkey300 committed:
Correct grammar in various javadocs to improve readability

otter502 committed:
Merge branch 'autos' into swerve

Sloth4915 committed:
Add method to orient towards nearest POI

Sloth4915 committed:
Merge branch 'bling' into autos

otter502 committed:
Fix issue where auto align went in the wrong direction when stationary

otter502 committed:
Fix issue where auto align went in the wrong direction when stationary

otter502 committed:
Merge branch 'autos' of https://github.com/Spartronics4915/2025-Reefscape into autos

otter502 committed:
Merge branch 'autos' into dynamics

otter502 committed:
Remove in real world error

otter502 committed:
Switch arm's primary rotation direction, slow down arm & elevator, add elevator presets

sharkey300 committed:
Add delay after scoring parameter to variable autos, set through complex auto chooser

otter502 committed:
Slow down motion profiles and confirm dynamics is working!

sharkey300 committed:
Increase arm/elevator constraints, arm kP, and elevator current limit to good enough values

sharkey300 committed:
Set L3/L4 elevator heights to values closer to current gartronics setup

sharkey300 committed:
Merge branch 'autos' of https://github.com/Spartronics4915/2025-Reefscape into autos

sharkey300 committed:
Update elastic dashboard config to include delay after scoring

sharkey300 committed:
Add autos for testing during gartronics

sharkey300 committed:
Half the max vel/acc of auto paths and auto align for safety during gartronics

sharkey300 committed:
Add JVM argugemnts to support profiling with VisualVM

sharkey300 committed:
Remove unneeded .gitkeep file from subsystems/coral/

sharkey300 committed:
Format control map and add unbinded buttons

sharkey300 committed:
Change comparison on isArmStowed to account for new CS

otter502 committed:
Remove VisualVM code, reverse intake speeds

otter502 committed:
Merge branch 'dynamics' of https://github.com/Spartronics4915/2025-Reefscape into dynamics

otter502 committed:
Increase intake intake Speed

otter502 committed:
Lower tolerances

otter502 committed:
Update elevator max height

otter502 committed:
Update minimum safe elevator height

otter502 committed:
Increase lasercan score trigger debounce

otter502 committed:
Add stow on coral intake trigger

otter502 committed:
Increase Dynamic presets for testbed values

otter502 committed:
Make intake commands not require subsystem

otter502 committed:
Fix Spelling for intake smartdashboard commands

sharkey300 committed:
Add gartronics dynamics scoring testing autos that score L3 and L4, currently not working

otter502 committed:
Fix transitions between L4 and load

otter502 committed:
Add in test preset for L2

otter502 committed:
Merge branch 'dynamics' of https://github.com/Spartronics4915/2025-Reefscape into dynamics

otter502 committed:
Fix safeToMove parallel movement by adding a waituntil safe to move

otter502 committed:
Remove strange delay when moving from prescore to L4

otter502 committed:
Update blocking intake method to also account for coral in the arm

otter502 committed:
Make intake command not run if there's already coral in the arm

otter502 committed:
Add javadoc for intake() command

sharkey300 committed:
Add image of car with ribbon to deploy directory

This is an inside joke within the drive team and can be removed if this causes issues.

otter502 committed:
Enable VisualVM

otter502 committed:
Edit Mechanism Render to properly render arm angle

otter502 committed:
Add logging and fix dynamics issue with parallel movement

otter502 committed:
Add elastic layout with manual controls

otter502 committed:
Fix delays and modify presets

otter502 committed:
Add L2 score button

otter502 committed:
Add force Arm Movement

otter502 committed:
Fix conditional logic

otter502 committed:
Modify manual control elastic layout

otter502 committed:
Make makeSystemSafeToMove command much easier to read and debug

otter502 committed:
Have the auto stow on intake not trigger in autos

otter502 committed:
Add intake button to smartdashboard

otter502 committed:
Add stow to start of debug autos

otter502 committed:
Add L2 to branch height

otter502 committed:
Add auto specific scoring

otter502 committed:
Remove ROI config application in intakeSubsystem

sharkey300 committed:
Move to last inputted score position when robot auto aligns to reef

otter502 committed:
Fix gartronics dynamics scoring points, and L2

otter502 committed:
Merge branch 'dynamics' of https://github.com/Spartronics4915/2025-Reefscape into dynamics

sharkey300 committed:
Add arm/elevator manual control to SmartDashboard

sharkey300 committed:
Add L1 scoring preset that doesn't work

sharkey300 committed:
Add L2 to operator controller and fix scoring button to work with new .score() method

sharkey300 committed:
Merge branch 'swerve' into dynamics

sharkey300 committed:
Tune comp chassis absolute encoder offsets and uninvert motors

sharkey300 committed:
Bind manual control for elevator and arm to d pad

otter502 committed:
Find encoder offsets for comp chassis

otter502 committed:
Add maximum height of elevator

otter502 committed:
Update L4 dynaPreset

otter502 committed:
Merge branch 'dynamics' of https://github.com/Spartronics4915/2025-Reefscape into dynamics

otter502 committed:
Increase debounce for intake

otter502 committed:
Remeasure encoder offsets with new magnet

otter502 committed:
Update gear ratios for comp chassis

sharkey300 committed:
Remove javadoc from getVisionMeasurement(SwerveSubsystem swerve)

It is tedious and unnecessary to maintain

sharkey300 committed:
Name Dynamics commands to distinguish them in the CommandScheduler

sharkey300 committed:
Name input commands to distinguish them in the CommandScheduler

sharkey300 committed:
Update README with D-pad controls and correct driver A/B buttons

sharkey300 committed:
Correct capitalization inconsistency in README

otter502 committed:
Move dynamics command factory file to commands folder

otter502 committed:
Merge branch 'dynamics' of https://github.com/Spartronics4915/2025-Reefscape into dynamics

otter502 committed:
Increase number of auto segments

otter502 committed:
Fix swerve drive angle drive motor issues

sharkey300 committed:
Add kGender constant to make the robot canonically agender

This is a joke and can be reverted if this causes problems.

sharkey300 committed:
Suppress the unused warning of the kGender variable

sharkey300 committed:
Reset comp chassis configuration to tune swerve drive

sharkey300 committed:
Set inverted status of drive and angle for comp chassis

sharkey300 committed:
Set module locations for comp chassis

sharkey300 committed:
Set conversion factors for comp chassis

The recorded diameter is converted to inches from the measured 8.81cm

sharkey300 committed:
Set absolute encoder offsets for comp chassis

sharkey300 committed:
Change stow angle and safe elev height as current settings hit the funnel

This is only needed because the arm is too close to the elevator. If the arm position is corrected, revert this commit.

otter502 committed:
Update minimum safe elevator height

otter502 committed:
Update Tag offsets to be asymmetrical

otter502 committed:
Switch pigeon to Pigeon 2 in yagsl config

otter502 committed:
Update Lasercan debounce, constraints, and tolerances

otter502 committed:
Add L3 to auto chooser

otter502 committed:
Remove lastinputtedScore system

otter502 committed:
Adjust angle for load

otter502 committed:
Adjust Branch side tag offsets

otter502 committed:
Remove unneeed import

otter502 committed:
Change BEFORE_MATCH boolean in vision algorithm to check for teleop

This won't change anything for actual competition, but it'll make testing easier as we don't have to briefly enable autonomous every time we redeploy or power cycle the robot.

sharkey300 committed:
Add setPercentage to IntakeSubsystem and change code to use it instead of ClosedLoopController

sharkey300 committed:
Add intakePercentage to IntakeSpeed enum and use it in setPercentage()

sharkey300 committed:
Lower intake laserCAN debounce to work better with .set() for controlling intake

sharkey300 committed:
Change intake mode switch to use intakeMotors() so it works with setPercentage()

sharkey300 committed:
Merge branch 'dynamics' into vision

sharkey300 committed:
Log single tag pose ambiguity to SmartDashboard so we can see how it changes

sharkey300 committed:
Merge branch 'dynamics' into autos

# Conflicts:
#	src/main/java/com/spartronics4915/frc2025/commands/VariableAutos.java

sharkey300 committed:
Wait until preset before scoring in variable auto cycles

sharkey300 committed:
Only log STPA if we are logging vision diagnostics

sharkey300 committed:
Reject pose if it's before the match and we can't get a good heading

This is because we want to ensure we have a good starting pose when starting autos. By having this, we force drive team to position the robot properly.

sharkey300 committed:
Add L2 branch height to complex auto chooser

sharkey300 committed:
Remove gartronics autos and move create auto option to top

sharkey300 committed:
Merge branch 'vision' into autos

# Conflicts:
#	src/main/java/com/spartronics4915/frc2025/RobotContainer.java

sharkey300 commented on src/main/java/com/spartronics4915/frc2025/Constants.java:
```diff
@@ -179,17 +194,17 @@ public enum PathplannerConfigs{
-                    new Translation2d(Inches.of(12.25), Inches.of(-12.3125)),
-                    new Translation2d(Inches.of(-12.25), Inches.of(12.3125)),
-                    new Translation2d(Inches.of(-12.25), Inches.of(-12.3125))
+                    new Translation2d(Inches.of(12.625), Inches.of(12.5625)),
+                    new Translation2d(Inches.of(12.125), Inches.of(-12.5)),
+                    new Translation2d(Inches.of(-12), Inches.of(12.5)),
+                    new Translation2d(Inches.of(-12.125), Inches.of(-12.4375))
```
Change these to the comp chassis config

otter502 replied:
Has this been fixed or does it still need to be fixed?

sharkey300 replied:
This still needs to get fixed

sharkey300 commented on src/main/java/com/spartronics4915/frc2025/Constants.java:
```diff
@@ -263,10 +294,10 @@ public static final class ArmConstants {
-            SCORE(Rotation2d.fromDegrees(30)),
-            EH(Rotation2d.fromDegrees(90)),
-            STOW(Rotation2d.fromDegrees(180));
+            EH(Rotation2d.fromDegrees(270)),
```
What is "EH"?

otter502 replied:
Debug preset, can be deleted if needed

sharkey300 commented on src/main/java/com/spartronics4915/frc2025/Constants.java:
```diff
@@ -277,27 +308,34 @@ private ArmSubsystemState(Rotation2d angle) {
         public static final int kArmMotorID = 11;
-        public static final int kPositionConversionFactor = 1;
-        public static final int kVelocityConversionFactor = 1;
+        public static final double kPositionConversionFactor = 1 * 0.03888888888888;
+        public static final double kVelocityConversionFactor = 1 * 0.03888888888888;
```
Why do we multiply by 0.03888...?

otter502 replied:
This is the gear ratio for the arm

sharkey300 commented on src/main/java/com/spartronics4915/frc2025/commands/drive/ChassisSpeedSuppliers.java:
```diff
@@ -245,24 +252,49 @@ public static Rotation2d getFieldAngleBetween(Translation2d from, Translation2d
 
     //#endregion
 
+    //#region debug
+
+    private static boolean resetHeading = true;
+
+    private static boolean shouldResetHeading() {
+        return resetHeading;
+    }
+
+    public static void toggleResetHeading() {
+        resetHeading = !resetHeading;
+    }
+
+    //#endregion
```
We frankly could remove all of this -- it's not even used in RobotContainer anymore since all debug controls got wiped during a merge somewhere.

sharkey300 replied:
If you want to do this just push a change; don't make an issue.

sharkey300 replied:
This is very low priority and does not need to be completed for this merge request to go through

sharkey300 commented on src/main/java/com/spartronics4915/frc2025/subsystems/bling2/DriverCommunication.java:
```diff
@@ -0,0 +1,103 @@
+        Region closest = null;
+        double closestDistance = Double.MAX_VALUE;
+        for (Region reg : Region.values()) {
+            for (Translation2d position : (DriverStation.getAlliance().isPresent() ? (DriverStation.getAlliance().get().equals(Alliance.Blue) ? reg.bluePositions : reg.redPositions) : reg.redPositions)) {
```
@FinnSpartronics you can use Optional.getOrElse() here

sharkey300 replied:
This is small and is not necessary for the merge request

sharkey300 commented on src/main/java/com/spartronics4915/frc2025/subsystems/coral/ArmSubsystem.java:
```diff
@@ -35,10 +39,10 @@ public class ArmSubsystem extends SubsystemBase implements ModeSwitchInterface{
     private Rotation2d mCurrentSetPoint = Rotation2d.fromRotations(0);;
```
This should be initialized to 270 degrees. I've noticed that the arm jolts a bit when the robot is first enabled because of this.

sharkey300 replied:
This has been fixed in [3960d7e](https://github.com/Spartronics4915/2025-Reefscape/pull/86/commits/3960d7e0906033717ce9221dfe03095be62f4e88).

sharkey300 reviewed (commented):
I've added my thoughts, though someone else should also review this.

sharkey300 committed:
Initalize current set point to starting angle and move starting angle to constants

otter502 commented on build.gradle:
A quick idea: implement the auto-logging functionality after an out-of-error crash
We should look into it

sharkey300 replied:
See #91

otter502 commented on src/main/deploy/bling/funrainbow21.bling:
What exactly is this file?
@FinnSpartronics

sharkey300 replied:
That's the bling config file for the rainbow light show. You can check it out in sim if you'd like.

otter502 commented on src/main/deploy/pathplanner/settings.json:
We should tune these in the autos branch

otter502 replied:
This will happen in #88

sharkey300 replied:
See #88

sharkey300 commented:
That's the bling config file for the rainbow light show. You can check it out in sim if you'd like.

otter502 commented on src/main/java/com/spartronics4915/frc2025/commands/autos/AlignToReef.java:
```diff
@@ -131,7 +89,7 @@ public Command generateCommand(BranchSide side) {
 
 
-    public Command generateCommand(ReefSide reefTag, BranchSide side) {
+    public Command generateCommand(final ReefSide reefTag, BranchSide side) {
```
I made this change but I am unsure if it's necessary

sharkey300 replied:
If it doesn't cause issues then its fine.

otter502 commented on src/main/java/com/spartronics4915/frc2025/commands/autos/AlignToReef.java:
```diff
@@ -170,9 +127,9 @@ private Command getPathFromWaypoint(Pose2d waypoint) {
      */
     private Rotation2d getPathVelocityHeading(ChassisSpeeds cs, Pose2d target){
-        if (getVelocityMagnitude(cs).in(MetersPerSecond) < 0.01 ) {
+        if (getVelocityMagnitude(cs).in(MetersPerSecond) < 0.05 ) {
```
These values should really be constants

otter502 replied:
There are other magic numbers in this file that should be constants

otter502 replied:
See #90

otter502 commented on src/main/java/com/spartronics4915/frc2025/subsystems/coral/ArmSubsystem.java:
```diff
@@ -32,13 +36,13 @@ public class ArmSubsystem extends SubsystemBase implements ModeSwitchInterface{
+    private final DoublePublisher appliedOutPub = NetworkTableInstance.getDefault().getTable("logArm").getDoubleTopic("applied out").publish();
+    private final StructPublisher<Rotation2d> positionPub = NetworkTableInstance.getDefault().getTable("logArm").getStructTopic("position", Rotation2d.struct).publish();
+    private final StructPublisher<Rotation2d> desiredStatePub = NetworkTableInstance.getDefault().getTable("logArm").getStructTopic("desiredState", Rotation2d.struct).publish();
+    private final StructPublisher<Rotation2d> setpointpub = NetworkTableInstance.getDefault().getTable("logArm").getStructTopic("setpointpub", Rotation2d.struct).publish();
```
Do we want to remove these or put them in a more general format? (Such as nesting these and other similar logging functions under a general table?)

sharkey300 replied:
I would very much appreciate that. A Logging/Arm table would be good for here. Please capitalize words when making these!!! ("logging" bad, "Logging" good)

otter502 replied:
See #89

sharkey300 replied:
See #89

otter502 commented on src/main/java/com/spartronics4915/frc2025/subsystems/coral/ElevatorSubsystem.java:
```diff
@@ -73,11 +80,19 @@ public ElevatorSubsystem() {
 
-        FFCalculator = new ElevatorFeedforward(ElevatorConstants.kS, ElevatorConstants.kG, ElevatorConstants.kV, ElevatorConstants.kA);
+        // If you are setting constants to zero, please explicitly set them to zero in the code.
+        // It makes it *MUCH* easier to debug if you can quickly see which features and constants are being used.
```
We might want to implement or reattach the feed forward constants here.

sharkey300 replied:
Create an issue if we should do this, otherwise resolve this conversation

otter502 reviewed (changes requested):
Handful of changes / things we should take a look at, once the changes have been looked at / considered everything else looks good and ready to merge

otter502 committed:
Add a warm up command to robotInit

otter502 committed:
Seperate driveController into a seperate constant

otter502 committed:
Write Position PID command

This command runs a pid loop for a specificed amount of time or until the swerve is within some tolerance of the goal state

otter502 committed:
Add PositionPID to the autoAlign command

otter502 committed:
Merge branch 'autos' into position-pid-command

otter502 commented:
We now have issues with all of these items, There aren't any outright things that need code changes immediately, other than the pathplanner module locations
Once that's changed I will approve this PR

otter502 committed:
Add debug print commands for testing

sharkey300 committed:
Lower intake scoring debounce so scoring is much faster

sharkey300 committed:
Double max robot speed so robot drives faster

sharkey300 committed:
Add commented out button to attempt to unjam coral

This still has issues!

sharkey300 committed:
Change coral station alignment to point funnel towards station, not arm

sharkey300 committed:
Disable limelights 12 and 13 as they cause more issues than its worth

otter502 committed:
Merge branch 'autos' into position-pid-command

otter502 committed:
Lower tolerances

otter502 committed:
Adjust auto align positions

otter502 committed:
Switch positoin pid to trigger

otter502 committed:
update paths to be closer to the Station

otter502 committed:
Increase adjustment tolerance & kp for arm

otter502 committed:
Add nudge buttons

otter502 committed:
Adjust align to reef minimum velocity compensation magnitude

This is the mangitude below which the velocity isn't compensated for by the swerve drive

otter502 committed:
Reduce debounce for end trigger

otter502 committed:
Add logging for the end trigger of the positionPIDCommand

sharkey300 committed:
Swap terms in vector subtraction when determining aligntoreef path velocity heading

This change was suggested by otter502 and supposedly improves auto align performance due to sign conventions or something

sharkey300 committed:
Log speed of swerve drive to NetworkTables

sharkey300 committed:
Add check for if robot speed is below a threshold for ending PositionPIDCommand

otter502 committed:
Sequence stow with the return path

otter502 committed:
Have autos stow when they get a coral

otter502 committed:
Add logging to the adjustment pid, and have debounce be a constant

otter502 committed:
increase position tolerance slightly

otter502 committed:
Reduce complex auto chooser steps

otter502 committed:
Increase auto's max speed

otter502 committed:
Have starting auto cycle not stow at the start

otter502 committed:
Add climb preset

otter502 committed:
Modify debounces and timeouts for auto align

otter502 committed:
Add setPreset1 command to smartdashboard

otter502 committed:
Adjust auto align's tag offsets

otter502 committed:
Have adjustment command lock the swerve drive after it finishes

otter502 committed:
Add some text to clarify the println

otter502 committed:
Add stow statements so that, during auto, it'll goto prescore

otter502 reviewed (approved):
looks good!
